editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,450 @@
+
+	if (remove_check_box->is_pressed()) {
+		undo_redo->add_do_method(this, "remove_references", name);
+	}
+
+	undo_redo->add_do_method(this, "call_deferred", "update_groups");
+	undo_redo->add_undo_method(this, "call_deferred", "update_groups");
+
+	undo_redo->add_do_method(this, "emit_signal", group_changed);
+	undo_redo->add_undo_method(this, "emit_signal", group_changed);
+
+	undo_redo->commit_action();
+}
+
+void GroupSettingsEditor::show_message(const String &p_message) {
+	message->set_text(p_message);
+	message->popup_centered();
+}
+
+void GroupSettingsEditor::_add_button_pressed() {
+	_add_group(group_name->get_text(), group_description->get_text());
+}
+
+void GroupSettingsEditor::_changed() {
+	emit_signal(group_changed);
+	call_deferred("update_groups");
+}
+
+void GroupSettingsEditor::_show_remove_dialog() {
+	if (!remove_dialog) {
+		remove_dialog = memnew(ConfirmationDialog);
+		remove_dialog->connect("confirmed", callable_mp(this, &GroupSettingsEditor::_confirm_delete));
+
+		VBoxContainer *vbox = memnew(VBoxContainer);
+		remove_label = memnew(Label);
+		vbox->add_child(remove_label);
+
+		remove_check_box = memnew(CheckBox);
+		remove_check_box->set_text(TTR("Delete references from all scenes"));
+		vbox->add_child(remove_check_box);
+
+		remove_dialog->add_child(vbox);
+
+		add_child(remove_dialog);
+	}
+
+	TreeItem *ti = tree->get_selected();
+	if (!ti) {
+		return;
+	}
+
+	remove_check_box->set_pressed(false);
+	remove_label->set_text(vformat(TTR("Delete group \"%s\" and its description?"), ti->get_text(0)));
+
+	remove_dialog->reset_size();
+	remove_dialog->popup_centered();
+}
+
+GroupSettingsEditor::GroupSettingsEditor() {
+	HBoxContainer *hbc = memnew(HBoxContainer);
+	add_child(hbc);
+
+	Label *l = memnew(Label);
+	l->set_text(TTR("Name:"));
+	hbc->add_child(l);
+
+	group_name = memnew(LineEdit);
+	group_name->set_h_size_flags(SIZE_EXPAND_FILL);
+	group_name->set_clear_button_enabled(true);
+	group_name->connect("text_changed", callable_mp(this, &GroupSettingsEditor::_group_name_text_changed));
+	hbc->add_child(group_name);
+
+	l = memnew(Label);
+	l->set_text(TTR("Description:"));
+	hbc->add_child(l);
+
+	group_description = memnew(LineEdit);
+	group_description->set_clear_button_enabled(true);
+	group_description->set_h_size_flags(SIZE_EXPAND_FILL);
+	hbc->add_child(group_description);
+
+	add_button = memnew(Button);
+	add_button->set_text(TTR("Add"));
+	add_button->set_disabled(true);
+	add_button->connect("pressed", callable_mp(this, &GroupSettingsEditor::_add_button_pressed));
+	hbc->add_child(add_button);
+
+	tree = memnew(Tree);
+	tree->set_hide_root(true);
+	tree->set_select_mode(Tree::SELECT_SINGLE);
+	tree->set_allow_reselect(true);
+
+	tree->set_columns(3);
+	tree->set_column_titles_visible(true);
+
+	tree->set_column_title(0, TTR("Name"));
+	tree->set_column_expand(0, true);
+
+	tree->set_column_title(1, TTR("Description"));
+	tree->set_column_expand(1, true);
+
+	tree->set_column_expand(2, false);
```

Comment:
```suggestion
	tree->set_column_title(0, TTR("Name"));
	tree->set_column_title(1, TTR("Description"));
	tree->set_column_expand(2, false);
```
Columns are expanded by default.